macros/Compat.ejs

```diff
@@ -1,66 +1,401 @@
+        for (let entry of support[browserName]) {
+          pushNotes(entry, browserName);
+        }
+      } else {
+          pushNotes(support[browserName], browserName);
+      }
+    }
+  }
+  return notesArray;
+}
+
+/*
+For a single row, write all the cells that contain support data.
+(That is, every cell in the row except the first, which contains
+an identifier for the row,  like "Basic support".
+
+`json_subfeature` is a `subfeature` JSON object.
+*/
+function writeSupportCells(json_subfeature, compatNotes, browserNames) {
+  let output = "";
+
+  for (let browserNameKey of Object.keys(browserNames)) {
+    let supportData = json_subfeature["support"][browserNameKey];
+    let supportInfo = '';
+    // if supportData is an array, there are multiple support statements
+    if (Array.isArray(supportData)) {
+      for (let entry of supportData) {
+        supportInfo += `<p>${getSupportInfo(entry, browserNameKey, compatNotes)}</p>`;
+      }
+    } else if (supportData) { // there is just one support statement
+      supportInfo = getSupportInfo(supportData, browserNameKey, compatNotes);
+    } else { // this browser has no info, it's unknown
+      supportInfo = getSupportInfo(null);
     }
+    output += `<td>${supportInfo}</td>`;
+  }
+  return output;
+}
+
+/*
+Generate a table that contains the detailed compat data for a single feature.
+The first row will contain the "basic_support" data.
+Any subsequent rows will contain "sub-feature" support data.
+
+`json_subfeature_set` is a `subfeature_set` JSON object.
+*/
+function writeFeatureTable(json_subfeature_set, environment, browserNames) {
+  let compatNotes = collectCompatNotes(json_subfeature_set);
+  let output = writeTableHead(environment, browserNames);
+  output += "<tbody>";
+  // write basic support first
+  output += `<tr><td>${localize(compatStrings, 'feature_basicsupport')}</td>`;
+  output += `${writeSupportCells(json_subfeature_set["basic_support"], compatNotes, browserNames)}</tr>`;
+  // then all the other sub-features, one per row
+  for (let row of Object.keys(json_subfeature_set)) {
+    if (row != "basic_support") {
+      var desc = json_subfeature_set[row].desc || row;
+      output += `<tr><td><code>${desc}</code></td>`
+      output += `${writeSupportCells(json_subfeature_set[row], compatNotes, browserNames)}</tr>`;
+    }
+  }
+  output += "</tbody></table></div>";
+  return output;
 }
 
-if (output === '') {
-  output = noDataFoundl10n;
+/*
+ Write each compat note, with an `id` so it will be linked from the table.
+*/
+function writeNotes(json_subfeature_set) {
+  let output = '';
+  let compatNotes = collectCompatNotes(json_subfeature_set);
+  for (let note of compatNotes) {
+    let noteIndex = compatNotes.indexOf(note);
+    output += `<p id=compatNote_${noteIndex+1}>${noteIndex+1}. ${note}</p>`;
+  }
+  return output;
 }
 
+/*
+Get compat data using sing a query string like "webextensions.api.alarms"
+*/
+function getData(queryString, obj) {
```

Comment:
Thanks :) Very happy about the query strings. Feels amazing to query the data like this.